### PR TITLE
Add: Release workflows for Cloud and 3rd Gen, Helm build-push

### PR DESCRIPTION
**Pull request by y0urself**
## What
Add: Release workflows for Cloud and 3rd Gen, Helm build-push

## Why
Add generic workflows to call from different repos

## References
DEVOPS-672

### CODE REVIEW

File: .github/workflows/helm-build-push.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      fail-fast: false
+      matrix:
+        chart:
+          [ split(${{ github.event.inputs.charts }}, ',') ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get version from tag
+        shell: bash
```
Comment (bjoernricks) [inline .github/workflows/helm-build-push.yml:23]:
```suggestion
        shell: bash
        id: version
```

File: .github/workflows/helm-build-push.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        run: |
+          vtag='${{github.ref_name}}'
+          echo "TAG=${vtag:1}" >> $GITHUB_ENV
```
Comment (bjoernricks) [inline .github/workflows/helm-build-push.yml:26]:
```suggestion
          echo "tag=${vtag:1}" >> $GITHUB_OUTPUT
```

File: .github/workflows/helm-build-push.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+      - name: Run helm version upgrade
+        uses: greenbone/actions/helm-version-upgrade@v2
+        with:
+          chart-path: ${{ github.workspace }}/charts/${{ matrix.chart }}
+          chart-version: ${{ env.TAG }}
```
Comment (bjoernricks) [inline .github/workflows/helm-build-push.yml:32]:
```suggestion
          chart-version: ${{ steps.version.outputs.tag }}
```

File: .github/workflows/release-cloud.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        run: |
+          if [[ "${{ github.event_name }}" = "workflow_dispatch" ]]; then
+            echo "RELEASE_REF=${{ github.ref_name }}" >> $GITHUB_ENV
+          else
+            echo "RELEASE_REF=${{ github.base_ref }}" >> $GITHUB_ENV
+          fi
+      - name: Release with release action
+        uses: greenbone/actions/release@v2
+        with:
+          python-version: "3.10"
```
Comment (bjoernricks) [inline .github/workflows/release-cloud.yml:73]:
```suggestion
```

File: .github/workflows/release-3rd-gen.yml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        run: |
+          if [[ "${{ github.event_name }}" = "workflow_dispatch" ]]; then
+            echo "RELEASE_REF=${{ github.ref_name }}" >> $GITHUB_ENV
+          else
+            echo "RELEASE_REF=${{ github.base_ref }}" >> $GITHUB_ENV
+          fi
+      - name: Release with release action
+        uses: greenbone/actions/release@v2
+        with:
+          python-version: "3.10"
```
Comment (bjoernricks) [inline .github/workflows/release-3rd-gen.yml:73]:
```suggestion
```

File: .github/workflows/helm-build-push.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+jobs:
+  release-helm-chart:
+    name: Release helm chart
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        chart:
+          [ split(${{ github.event.inputs.charts }}, ',') ]
```
Comment (bjoernricks) [inline .github/workflows/helm-build-push.yml:17]: I am looking forward if this works at the end :-)

File: .github/workflows/helm-build-push.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        run: |
+          cat '${{ github.workspace }}/charts/${{ matrix.chart }}/Chart.yaml'
+
+      - name: Upload to github registry
+        uses: greenbone/actions/helm-build-push@v2
+        with:
+          chart-name: ${{ matrix.chart }}
+          registry: ${{ vars.IMAGE_REGISTRY }}
+          registry-subpath: helm-charts/
+          registry-user: ${{ secrets.GREENBONE_BOT }}
```
Comment (bjoernricks) [inline .github/workflows/helm-build-push.yml:44]: IMHO secrets must be handled specially in a reusable workflow https://docs.github.com/en/actions/using-workflows/reusing-workflows#creating-a-reusable-workflow

File: .github/workflows/helm-build-push.yml (hunk 6 above)
Comment (y0urself) [reply .github/workflows/helm-build-push.yml:18]: Me too :D

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**